frontend/JavaScript/index.js

```diff
@@ -0,0 +1,296 @@
+  catLevel2 = urlParams.get('cat2');
+  urlSortKey = urlParams.get('sort')
+
+  if (prod_query!=null){
+    window.parent.location = `index.html?q=${prod_query}&page=1&sort=${sortKey}`
+  }
+
+  else if (catLevel1!=null ){
+    window.parent.location = `index.html?cat1=${catLevel1}&cat2=${catLevel2}&page=${pageNumber}&sort=${sortKey}`
+  }
+
+    else {
+      window.parent.location = `index.html?q=*&page=1&sort=${sortKey}`
+  }
+}
+
+window.onload=function reload(){
+  const queryString = window.location.search;
+  const urlParams = new URLSearchParams(queryString);
+  
+  var prod_query =urlParams.get('q');
+  if(prod_query==""){
+    var prod_query='*';
+  }
+  var catLevel1 =urlParams.get('cat1');
+  var catLevel2 =urlParams.get('cat2');
+  var pageNumber = urlParams.get('page')
+  var sortKey = urlParams.get('sort')
+  
+  if (pageNumber==null){
+    pageNumber = 1;
+  }
+
+  if (sortKey==null){
+    sortKey = '';
+  }
+
+  fetch(`http://127.0.0.1:7002/categoryTree`,{
+    method : 'GET',
+    mode :'cors',
+    headers:{
+      'Access-Control-Allow-Origin':'*',
+      'Accept': 'application/json',
+      'Content-Type': 'application/json;charset=utf-8',
+      
+    }
+    
+  }).then(response => response.json()).then(data =>{
+    cat1 = data[0];
+    cat2 = data[1];
+    cat3 = data[2];
+    var cat_container=document.getElementById("dropdownheader");
+    cat_container.innerHTML =``;
+    for( let i = 0; i < cat1.length; i++){
+      var catId = cat1[i][1];
+      var catValue = cat1[i][0];
+      var value = cat1[i][0];
+      if (catValue!='exp'){
+      dataElement = data[parseInt(catId)+1];
+      var elementArr = dataElement[catId].sort();
+      var tempo = ``;
+      for( let j = 0;j < elementArr.length; j++){
+        var valueName = elementArr[j].replace(/([A-Z])/g, ' $1').trim();
+        tempo += `<li onclick="getCategory('${catId}','${elementArr[j]}')" class="hover-button">${valueName}</li>`;
+      }
+      }
+      if (catValue=='exp'){
+        value = 'Others';
+        cat_container.innerHTML += `<div class="right-menu">
+        <button class="button" onclick="getCategory('${catId}','${value}')">Gift Cards</button>
+        </div>`;
+      }
+      else{
+      cat_container.innerHTML += `<div class="right-menu">
+      <button class="button" onclick="getCategory('${catId}',value)" value="${value}">${catValue}</button>
+      <div class="dropdown-menu">` + tempo;
+      }
+    }
+  });
+  if (prod_query!=null){
+    document.getElementById('loading').style.display='block';
+    setTimeout(()=>{
+    fetch(`http://127.0.0.1:7002/product_query?q=${prod_query}&page=${pageNumber}&sort=${sortKey}`,{
+    method : 'GET',
+    mode :'cors',
+    headers:{
+      'Access-Control-Allow-Origin':'*',
+      'Accept': 'application/json',
+      'Content-Type': 'application/json;charset=utf-8',
+      
+    }
+    
+  }).then(response => response.json()).then(data =>{
+      var pages = data[0]
```

Comment:
Add null exceptions

Reply:
Null exceptions has been handled for all the .js files.